1.3.0.Final quarkus:list-extensions doesn't work in non-quarkus project directory

1.3.0.Final quarkus:list-extensions doesn't work in non-quarkus project directory
`mvn io.quarkus:quarkus-maven-plugin:1.2.1.Final:list-extensions` works in any directory

Steps to reproduce:
```
mkdir foo && cd foo
mvn io.quarkus:quarkus-maven-plugin:1.3.0.Final:list-extensions
```

```
mvn io.quarkus:quarkus-maven-plugin:1.3.0.Final:list-extensions
[INFO] Scanning for projects...
...
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:1.3.0.Final:list-extensions 
  (default-cli) on project standalone-pom: Failed to list extensions: Neither project's 
  build file handler nor the project writer has been provided -> [Help 1]

```

```
Caused by: org.apache.maven.plugin.MojoExecutionException: Failed to list extensions
    at io.quarkus.maven.ListExtensionsMojo.doExecute (ListExtensionsMojo.java:50)
    at io.quarkus.maven.BuildFileMojoBase.execute (BuildFileMojoBase.java:120)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
...
Caused by: java.lang.IllegalStateException: Neither project's build file handler nor the project writer has been provided
    at io.quarkus.cli.commands.QuarkusCommandInvocation.getBuildFile (QuarkusCommandInvocation.java:93)
    at io.quarkus.cli.commands.QuarkusCommandInvocation.getBuildFile (QuarkusCommandInvocation.java:86)
    at io.quarkus.cli.commands.ListExtensionsCommandHandler.execute (ListExtensionsCommandHandler.java:69)
    at io.quarkus.cli.commands.ListExtensions.execute (ListExtensions.java:73)
    at io.quarkus.maven.ListExtensionsMojo.doExecute (ListExtensionsMojo.java:48)
    at io.quarkus.maven.BuildFileMojoBase.execute (BuildFileMojoBase.java:120)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
```

--- a/independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensionsCommandHandler.java
+++ b/independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/ListExtensionsCommandHandler.java
@@ -66,28 +66,39 @@ public class ListExtensionsCommandHandler implements QuarkusCommand {
                     currentFormatter = this::conciseFormatter;
             }
 
-            final BuildFile buildFile = invocation.getBuildFile();
+            final BuildFile buildFile = getBuildFileSafe(invocation);
             loadedExtensions.forEach(extension -> display(extension, installed, all, currentFormatter, buildFile));
 
-            if ("concise".equalsIgnoreCase(format)) {
-                if (buildFile instanceof GradleBuildFile) {
-                    System.out.println("\nTo get more information, append --format=full to your command line.");
-                } else {
-                    System.out
-                            .println("\nTo get more information, append -Dquarkus.extension.format=full to your command line.");
+            if (buildFile != null) {
+                if ("concise".equalsIgnoreCase(format)) {
+                    if (buildFile instanceof GradleBuildFile) {
+                        System.out.println("\nTo get more information, append --format=full to your command line.");
+                    } else {
+                        System.out.println(
+                                "\nTo get more information, append -Dquarkus.extension.format=full to your command line.");
+                    }
                 }
-            }
 
-            if (buildFile instanceof GradleBuildFile) {
-                System.out.println("\nAdd an extension to your project by adding the dependency to your " +
-                        "build.gradle or use `./gradlew addExtension --extensions=\"artifactId\"`");
-            } else {
-                System.out.println("\nAdd an extension to your project by adding the dependency to your " +
-                        "pom.xml or use `./mvnw quarkus:add-extension -Dextensions=\"artifactId\"`");
+                if (buildFile instanceof GradleBuildFile) {
+                    System.out.println("\nAdd an extension to your project by adding the dependency to your " +
+                            "build.gradle or use `./gradlew addExtension --extensions=\"artifactId\"`");
+                } else {
+                    System.out.println("\nAdd an extension to your project by adding the dependency to your " +
+                            "pom.xml or use `./mvnw quarkus:add-extension -Dextensions=\"artifactId\"`");
+                }
             }
         }
 
         return QuarkusCommandOutcome.success();
+    }
+
+    private BuildFile getBuildFileSafe(QuarkusCommandInvocation invocation) {
+        try {
+            return invocation.getBuildFile();
+        } catch (Exception e) {
+            invocation.log.debug("Unable to get build file. Error is %s ", e.getMessage());
+            return null;
+        }
     }
 
     Map<String, Dependency> findInstalled(QuarkusCommandInvocation invocation) throws IOException {
@@ -151,7 +162,7 @@ public class ListExtensionsCommandHandler implements QuarkusCommand {
 
     private String extractVersion(final Dependency dependency, BuildFile buildFile) {
         String version = dependency != null ? dependency.getVersion() : null;
-        if (version != null && version.startsWith("$")) {
+        if (version != null && version.startsWith("$") && buildFile != null) {
             String value = null;
             try {
                 value = (String) buildFile.getProperty(propertyName(version));

